prow/cmd/mkpj/main.go

```diff
@@ -57,15 +139,24 @@ func (o *options) Validate() error {
 
 func gatherOptions() options {
 	o := options{}
-	flag.StringVar(&o.jobName, "job", "", "Job to run.")
-	flag.StringVar(&o.configPath, "config-path", "", "Path to config.yaml.")
-	flag.StringVar(&o.jobConfigPath, "job-config-path", "", "Path to prow job configs.")
-	flag.StringVar(&o.baseRef, "base-ref", "", "Git base ref under test")
-	flag.StringVar(&o.baseSha, "base-sha", "", "Git base SHA under test")
-	flag.IntVar(&o.pullNumber, "pull-number", 0, "Git pull number under test")
-	flag.StringVar(&o.pullSha, "pull-sha", "", "Git pull SHA under test")
-	flag.StringVar(&o.pullAuthor, "pull-author", "", "Git pull author under test")
-	flag.Parse()
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	fs.StringVar(&o.jobName, "job", "", "Job to run.")
+	fs.StringVar(&o.configPath, "config-path", "", "Path to config.yaml.")
+	fs.StringVar(&o.jobConfigPath, "job-config-path", "", "Path to prow job configs.")
+	fs.StringVar(&o.baseRef, "base-ref", "", "Git base ref under test")
+	fs.StringVar(&o.baseSha, "base-sha", "", "Git base SHA under test")
+	fs.IntVar(&o.pullNumber, "pull-number", 0, "Git pull number under test")
+	fs.StringVar(&o.pullSha, "pull-sha", "", "Git pull SHA under test")
+	fs.StringVar(&o.pullAuthor, "pull-author", "", "Git pull author under test")
+	o.github.AddFlags(fs)
+	// Do not default github-token-path to `/etc/github/oauth` to allow running
```

Comment:
/shrug

Reply:
Might be better to refactor the GitHubOptions to allow for an unauthenticated user

Reply:
I do understand that sentiment, the approach is not particularly elegant. Just removing the default inside will break a lot of installations I assume.

The only not completely ugly approach I can think of is changing the signature of `func (o *GitHubOptions) AddFlags(fs *flag.FlagSet)` to `func (o *GithubOptions) AddFlags(fs *flag.Flagset, wantDefaultForGithubTokenPath bool)`

WDYT?

Reply:
Or maybe `type PotentiallyUnatuthenticatedGitHubOptions struct`? :)

Reply:
A third option would also be `func (o *GithubOptions) AdddFlagsWithoutDefaultGithubTokenPath(fs *flag.Flagset)`

@cjwagner Any preference? Or just leave it as-is?

Reply:
I actually _love_ the last suggestion of optionally adding the different set of flags.

Reply:
Added that, PTAL